mlflow/sklearn/utils.py

```diff
@@ -103,6 +118,248 @@ def _get_args_for_score(score_func, fit_func, fit_args, fit_kwargs):
+    for metric in metrics_list:
+        try:
+            metric_value = metric.function(**metric.arguments)
+        except Exception as e:  # pylint: disable=broad-except
+            _log_warning_for_metrics(metric.name, metric.function, e)
+        else:
+            metric_value_dict[metric.name] = metric_value
+
+    return metric_value_dict
+
+
+def _get_classifier_metrics(trained_estimator, fit_args, fit_kwargs):
+    """
+    Compute and log various common metrics for classifiers
+
+    For (1) precision score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html#sklearn.metrics.precision_score
+    (2) recall score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html#sklearn.metrics.recall_score
+    (3) f1_score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score
+    By default, we choose the parameter `labels` to be `None`, `pos_label` to be `1`,
+    `average` to be `weighted` to compute the weighted precision score.
+
+    For (4) accuracy score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html
+    we choose the parameter `normalize` to be `True` to output the percentage of accuracy,
+    as opposed to `False` that outputs the absolute correct number of sample prediction
+
+    We log additional metrics if certain classifier has method `predict_proba`
+    (5) log loss:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html#sklearn.metrics.log_loss
+    (6) roc_auc_score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#r4bb7c4558997-5
+    By default, for roc_auc_score, we pick `average` to be `weighted`, `multi_class` to be `ovo`,
+    to make the output more insensitive to dataset imbalance.
+
+    Steps:
+    1. Extract X and y_true from fit_args and fit_kwargs, and compute y_pred.
+    2. If the sample_weight argument exists in fit_func (accuracy_score by default
+    has sample_weight), extract it from fit_args or fit_kwargs as
+    (y_true, y_pred, ...... sample_weight), otherwise as (y_true, y_pred, ......)
+    3. return a dictionary of metric(name, value)
+
+    :param trained_estimator: The already fitted classifier
+    :param fit_args: Positional arguments given to fit_func.
+    :param fit_kwargs: Keyword arguments given to fit_func.
+    :return: dictionary of (function name, computed value)
+    """
+    import sklearn
+
+    fit_arg_names = _get_arg_names(trained_estimator.fit)
+    X, y_true, y_pred = _get_samples_labels_and_predictions(
+        trained_estimator, fit_args, fit_kwargs, fit_arg_names
+    )
+    sample_weight = (
+        _get_sample_weight(fit_arg_names, fit_args, fit_kwargs)
+        if _SAMPLE_WEIGHT in fit_arg_names
+        else None
+    )
+
+    classifier_metrics = [
+        _SklearnMetric(
+            name="precision_score",
```

Comment:
```suggestion
            name=sklearn.metrics.precision_score.__name__
```

Can you de-hardcode `name` to prevent typos?